### ISSUE TITLE
3D-surface mesh

### ISSUE THREAD

**Issue by shengdie** [bfroehle/pydistmesh#14]
Is the 3d-surface mesh implemented? Can you give an example? Thank you.

**Comment by popersson** [comment]
No... should be pretty easy to do though, most of the work is in the C-code trisurfupd.cpp.

**Comment by shengdie** [comment]
Could you please consider adding this? Thank you.

> No... should be pretty easy to do though, most of the work is in the C-code trisurfupd.cpp.

**Comment by dhermes** [comment]
@shengdie This library isn't actively maintained unfortunately.

----

@popersson I found it here https://github.com/robEllenberg/MATLAB/blob/master/toolbox/distmesh/trisurfupd.cpp. I know you distribute it via http://persson.berkeley.edu/distmesh/distmesh.zip, is it also available somewhere on GitHub that **you** maintain?

**Comment by krober10nd** [comment]
Hello, I know this has been sitting here for some time. But if you're curious about surface meshing with this distmesh algorithm, I at one point implemented this (in MATLAB) using cubic Bezier surfaces as @popersson recommended I think in his thesis. https://github.com/krober10nd/SeismicMesh/blob/3dmeshing/Example_SurfaceMeshing.m

The problem with using the original surface meshing algorithm is that the initial point generation doesn't work well for non-uniform meshes. 

I also implemented trisurfpd in fortran at one point. If that's useful let me know. 

hope it helps

**Comment by krober10nd** [comment]
...but by using cubic Bezier surfaces and projecting them back and forth into the unit space and onto the 3D  surface, the problem can be ameliorated.
